src/snapshot_tests/test_case.rs

```diff
@@ -224,70 +234,126 @@ pub struct TestInput {
+        RGBColor(255, 165, 0),
         // WHITE, input_7
-        (255, 255, 255),
+        RGBColor(255, 255, 255),
         // GRAY, input_8
-        (128, 128, 128),
+        RGBColor(128, 128, 128),
         // LIGHT_RED, input_9
-        (255, 128, 128),
+        RGBColor(255, 128, 128),
         // LIGHT_BLUE, input_10
-        (128, 128, 255),
+        RGBColor(128, 128, 255),
         // LIGHT_GREEN, input_11
-        (128, 255, 128),
+        RGBColor(128, 255, 128),
         // PINK, input_12
-        (255, 192, 203),
+        RGBColor(255, 192, 203),
         // PURPLE, input_13
-        (128, 0, 128),
+        RGBColor(128, 0, 128),
         // BROWN, input_14
-        (165, 42, 42),
+        RGBColor(165, 42, 42),
         // YELLOW_GREEN, input_15
-        (154, 205, 50),
+        RGBColor(154, 205, 50),
         // LIGHT_YELLOW, input_16
-        (255, 255, 224),
+        RGBColor(255, 255, 224),
     ];
 
     pub fn new(index: usize) -> Self {
-        Self::new_with_resolution(
+        Self::new_patterned(index, TestInputPattern::Plain)
+    }
+
+    pub fn new_with_resolution(index: usize, resolution: Resolution) -> Self {
+        Self::new_patterned_with_resolution(index, TestInputPattern::Plain, resolution)
+    }
+
+    pub fn new_patterned(index: usize, pattern: TestInputPattern) -> Self {
+        Self::new_patterned_with_resolution(
             index,
+            pattern,
             Resolution {
                 width: 640,
                 height: 360,
             },
         )
     }
 
-    pub fn new_with_resolution(index: usize, resolution: Resolution) -> Self {
-        if index >= Self::COLOR_VARIANTS.len() {
-            panic!("Reached input amount limit: {}", Self::COLOR_VARIANTS.len())
-        }
-
+    pub fn new_patterned_with_resolution(
+        index: usize,
+        pattern: TestInputPattern,
+        resolution: Resolution,
+    ) -> Self {
         let color = Self::COLOR_VARIANTS[index];
-        let r = color.0 as f32;
-        let g = color.1 as f32;
-        let b = color.2 as f32;
+        let mut y_plane = vec![0; resolution.width * resolution.height];
+        let mut u_plane = vec![0; (resolution.width * resolution.height) / 4];
+        let mut v_plane = vec![0; (resolution.width * resolution.height) / 4];
+
+        let rgb_color = |x: usize, y: usize| match pattern {
+            TestInputPattern::Border => {
+                let border_size = resolution.width / 50;
+                let is_border_in_x = x <= border_size
+                    || (x <= resolution.width && x >= resolution.width - border_size);
+                let is_border_in_y: bool = y <= border_size
+                    || (y <= resolution.height && y >= resolution.height - border_size);
+
+                if is_border_in_x || is_border_in_y {
+                    RGBColor::BLACK
+                } else {
+                    color
+                }
+            }
+            TestInputPattern::Grid => {
```

Comment:
I prefer constant value instead of taking 10% of width, if you take a percent of other dimensions you don't know whether the input has a different size, or whether the component was rescalled.